api/core.py

```diff
@@ -101,10 +104,19 @@
     'additionalProperties': False
 }
 
-def _append_targets(targets, container, prefix, total_size, total_cnt, optional, data_path, attachments=True):
+def _append_targets(targets, container, prefix, total_size, total_cnt, optional, data_path, filter_=None):
     for f in container['files']:
-        if (not attachments or (type(attachments) == list and f['filename'] not in attachments)) and 'attachment' in f.get('tags', []):
-            continue
+        if filter_:
+            if not filter_['attachments'] and f['flavor'] == 'attachment':
+                continue
+            if not filter_['dicom'] and f['filetype'] == 'dicom':
+                continue
+            if not filter_['nifti'] and f['filetype'] == 'nifti':
+                continue
+            if not filter_['montage'] and f['filetype'] == 'montage':
+                continue
+            if not filter_['other'] and f['flavor'] != 'attachment' and f['filetype'] not in ['montage', 'nifti', 'dicom']:
+                continue
```

Comment:
filter_ is expected (if not null) to be  a json object with some boolean properties:
'attachments', 'dicom', 'nifti', 'montage', 'other'
Each of the if-conditions discards files if the corresponding filter properties is false.
e.g. if the 'attachments' property is false, discard the file if it is an attachment.
The last condition on 'other' excludes all the files that are not an attachment, a dicom, a nifti or a montage